jest/requireNativeComponent.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+'use strict';
+
+module.exports = () => {
+  const React = require('react');
+
+  const MockNativeMethods = require('./MockNativeMethods');
+
+  return viewName => {
+    class NativeComponent extends React.Component {
+      render() {
+        const {children, ...props} = this.props;
+
+        return React.createElement(viewName, props, children);
+      }
+    }
+
+    Object.assign(NativeComponent.prototype, {
```

Comment:
Why is this needed?

Reply:
This only mocks actual native components, so they will all need the native methods mocked out. And then for example, things like TextInput need _nativeTag and Animated (ScrollView) needs the viewConfig.